awx/main/tests/functional/test_inventory_source_injectors.py

```diff
@@ -191,7 +192,7 @@ def create_reference_data(source_dir, env, content):
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize('this_kind', CLOUD_PROVIDERS)
+@pytest.mark.parametrize('this_kind', discover_available_cloud_provider_plugin_names)
```

Comment:
This one is tricky. You cannot pass a callable to parametrize, it accepts iterables. Here's how you call it and it'll return a list that is something `parametrize` can work with:
```suggestion
@pytest.mark.parametrize('this_kind', discover_available_cloud_provider_plugin_names())
```

However, I suspect that this might be too early in the import cascade. If it is, this will need to be rewritten using `pytest-subtest` and called during testing, not the test collection stage.

I briefly mentioned this before (https://github.com/ansible/awx/pull/15537#discussion_r1782920489) as I suspected this might be a problem but haven't checked.